src/ILVerification/src/AccessVerificationHelpers.cs

```diff
@@ -252,18 +252,29 @@ private static bool CanAccessFamily(TypeDesc currentType, TypeDesc targetTypeDef
 
         private static bool GrantsFriendAccessTo(this ModuleDesc module, ModuleDesc friendModule)
         {
-            var assembly = (EcmaAssembly)module;
-            var friendName = ((IAssemblyDesc)friendModule).GetName();
-
-            foreach (var attribute in assembly.GetDecodedCustomAttributes("System.Runtime.CompilerServices", "InternalsVisibleToAttribute"))
+            var assembly = (module as EcmaAssembly);
+            if (assembly == null)
             {
-                AssemblyName friendAttributeName = new AssemblyName((string)attribute.FixedArguments[0].Value);
-                if (!friendName.Name.Equals(friendAttributeName.Name, StringComparison.OrdinalIgnoreCase))
-                    continue;
-
-                // Comparing PublicKeyToken, since GetPublicKey returns null due to a bug
-                if (IsSamePublicKey(friendAttributeName.GetPublicKeyToken(), friendName.GetPublicKeyToken()))
+                assembly = ((EcmaModule)module).ReferencingAssembly;
+            }
+            if (assembly != null)
+            {
+                if (assembly == friendModule)
+                {
                     return true;
+                }
+                var friendName = ((IAssemblyDesc)friendModule).GetName();
```

Comment:
`friendModule` can be netmodule too. This should use `ReferencingAssembly` as well.